plugins/modules/discovery_intent.py

```diff
@@ -490,6 +601,164 @@ def get_creds_ids_list(self):
         self.log("Credential Ids list passed is {0}".format(str(self.creds_ids_list)), "INFO")
         return self.creds_ids_list
 
+    def handle_global_credentials(self, response=None):
+        """
+        Method to convert values for create_params API when global paramters
+        are passed as input.
+
+        Parameters:
+            - response: The response collected from the get_all_global_credentials_v2 API
+
+        Returns:
+            - global_cred_dict  : The dictionary containing list of IDs of various types of
+                                    Global credentials.
+        """
+
+        global_credentials = self.validated_config[0].get("global_credentials")
+        cli_credentials_list = global_credentials.get('cli_credentials_list')
+        http_read_credential_list = global_credentials.get('http_read_credential_list')
+        http_write_credential_list = global_credentials.get('http_write_credential_list')
+        snmp_v2_read_credential_list = global_credentials.get('snmp_v2_read_credential_list')
+        snmp_v2_write_credential_list = global_credentials.get('snmp_v2_write_credential_list')
+        snmp_v3_credential_list = global_credentials.get('snmp_v3_credential_list')
+        net_conf_port_list = global_credentials.get('net_conf_port_list')
+        global_cred_dict = {}
```

Comment:
Can we name it as
global_credentails_all = {} 
instead of saying as dict..

Reply:
Done